plugins/meta/tuning/tuning.go

```diff
@@ -60,6 +66,14 @@ type IPAMArgs struct {
 	Mtu     *int               `json:"mtu,omitempty"`
 }
 
+// configToRestore will contain interface attributes that should be restored on cmdDel
+type configToRestore struct {
```

Comment:
It strikes me that this struct could be a part of `TuningConf`, which would remove the need to repeat all fields.

(and maybe part of `IPAMArgs` ?)

Reply:
@bboreham I've changed the code. Renamed `configToRestore` to `ifConfig` and made it a part of `IPAMArgs`. I chose `IPAMArgs` because `Promisc` field in that structure is a pointer to bool same as in `configToRestore`. In case of json, simple bool will be omitted if it's value will be `false` and `omitempty` were specified. And I believe that in case of future development or just to avoid ambiguity it would be better to save exact value of `Promisc` in backup file whether it is true or false. 

Which leads me to another question I was wondering regarding the `Promisc bool` in `TuningConf`.  In original [code](https://github.com/containernetworking/plugins/blob/7e2c95b9f88bf274b6fe973d24bc90f3ab2cfcf6/plugins/meta/tuning/tuning.go#L354) of `cmdAdd` there is
```
if tuningConf.Promisc != false {
	if err = changePromisc(args.IfName, true); err != nil {
		return err
	}
}
```

So if user will have a NIC with promiscuous mode enabled, and would like to switch promisc off using tuning (by setting promisc to false), then tuning plugin will not disable the promiscuous mode on that NIC as the code inside if statement will not be executed. Is this an expected behaviour, e.g. linked to problems with some particular NIC or driver? If not, then `Promisc bool` could be changed to pointer, alongside MAC and MTU variables, and therefore `ifConfig` could be used in both `IPAMArgs` and `TuningConf`. I think that this shouldn't be a part of this PR anyway, so this question is just out of curiosity.